### PR TITLE
test: Add tests for Dashboard Header and HeaderActionsDropdown components

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
Adds and enhances the test cases for the Dashboard Header and HeaderActionsDropdown components.

### TEST PLAN
All tests should pass

### CODE REVIEW

**Comment by geido** [comment]
@junlincc @rusackas

File: superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/HeaderActionsDropdown.test.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { render, screen } from 'spec/helpers/testing-library';
+import userEvent from '@testing-library/user-event';
+import fetchMock from 'fetch-mock';
+import HeaderActionsDropdown from '.';
+
+const mockedProps = {
```
Comment (yardz) [inline superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/HeaderActionsDropdown.test.tsx:26]: What do you think of using factory here?

File: superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/HeaderActionsDropdown.test.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+test('should open the dropdown', async () => {
+  render(<HeaderActionsDropdown {...mockedProps} />);
+  await openDropdown();
+  expect(await screen.findByRole('menu')).toBeInTheDocument();
+});
+
+test('should render the menu items in edit mode', async () => {
+  render(<HeaderActionsDropdown {...mockedProps} />);
+  await openDropdown();
```
Comment (yardz) [inline superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/HeaderActionsDropdown.test.tsx:104]: I usually put a test beforehand to make sure the elements weren't there before the action. So there is an idea of "before it didn't exist" and "now it exists".
This prevents any unwanted side effects from going unnoticed.

Comment (yardz) [reply superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/HeaderActionsDropdown.test.tsx:104]: Same thing for the other tests

File: superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/HeaderActionsDropdown.test.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  expect(screen.getByText('Save as')).toBeInTheDocument();
+});
+
+test('should NOT render the "Save Modal" menu item when user cannot save', async () => {
+  const cannotSaveProps = {
+    ...mockedProps,
+    userCanSave: false,
+  };
+  render(<HeaderActionsDropdown {...cannotSaveProps} />);
+  await openDropdown();
```
Comment (yardz) [inline superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/HeaderActionsDropdown.test.tsx:139]: Same thing here. First I guarantee that it exists and then I see that it no longer exists.

File: superset-frontend/src/dashboard/components/Header/Header.test.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+ * under the License.
+ */
+import React from 'react';
+import { render, screen } from 'spec/helpers/testing-library';
+import userEvent from '@testing-library/user-event';
+import fetchMock from 'fetch-mock';
+import { HeaderProps } from './types';
+import Header from '.';
+
+const mockedProps = {
```
Comment (yardz) [inline superset-frontend/src/dashboard/components/Header/Header.test.tsx:26]: factory?

File: superset-frontend/src/dashboard/components/Header/types.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
+import { Layout } from 'src/dashboard/types';
```
Comment (yardz) [inline superset-frontend/src/dashboard/components/Header/types.ts:1]: Nice!

File: superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/HeaderActionsDropdown.test.tsx (hunk 2 above)
Comment (geido) [reply superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/HeaderActionsDropdown.test.tsx:104]: I agree. This is handled in the test above `should render the menu items`. It's basically the opposite state. I'll check if I missed this elsewhere. Thanks!

File: superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/HeaderActionsDropdown.test.tsx (hunk 3 above)
Comment (geido) [reply superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/HeaderActionsDropdown.test.tsx:139]: I think here as well you are referring to the previous test `should render the "Save Modal" when user can save` which is testing the opposite case where the "Save Modal" does exist.

**Comment by rusackas** [comment]
Unit tests aren't passing, and needs a little rebase, but this is looking SUPER close to golden!

**Review by rusackas — APPROVED** [review]
LGTM!

**Pull request merged**